core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/MaxCountPropertyShape.java

```diff
@@ -48,6 +48,7 @@
 		super(id, connection, nodeShape, deactivated, parent, path);
 
 		this.maxCount = maxCount;
+		this.Expected = SimpleValueFactory.getInstance().createLiteral(maxCount);
```

Comment:
I've wondered if minCount and maxCount should have the min and max as expected, or if they should have a string with the "<" and ">" in it. For maxCount 3 the string would be "<= 3". What do you think?